src/__test__/components.test.js

```diff
@@ -0,0 +1,46 @@
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
+import NavBar from '../component/NavBar/NavBar';
+import CryptoContainer from '../component/Home/CryptoContainer';
+import CoinDetails from '../component/Details/CoinDetails';
+import { store } from '../app/store';
+
+describe('Testing for all components', () => {
+  it('Testing for the CryptoContainer Components', () => {
+    const UI = render(
+      <Provider store={store}>
+        <BrowserRouter>
+          <Routes>
+            <Route element={<CryptoContainer />} />
+          </Routes>
+        </BrowserRouter>
+      </Provider>,
+    );
+    expect(UI).toMatchSnapshot();
+  });
+
+  it('Testing for the coinDetails component', () => {
+    const UI = render(
+      <Provider store={store}>
+        <BrowserRouter>
+          <Routes>
+            <Route element={<CoinDetails />} />
+          </Routes>
+        </BrowserRouter>
+      </Provider>,
+    );
+
+    expect(UI).toMatchSnapshot();
+  });
+
+  it('Testing the NavBar component', () => {
+    const Navbar = render(
+      <BrowserRouter>
+        <NavBar />
+      </BrowserRouter>,
+    );
+
+    expect(Navbar).toMatchSnapshot();
+  });
+});
```

Comment:
- [ ] Great testing your components but you forgot to test the pure function of redux code like action creators so please try to test the pure function in redux code in order to cover those with tests and also to get to the new project 
- Screenshot of the requirements
<img width="825" alt="Screen Shot 2022-10-05 at 16 22 19" src="https://user-images.githubusercontent.com/61790587/194099236-83ae22fe-d37b-47cb-bb82-6d4420404bfb.png">

Reply:
Write test for async functions at pending and fulfilled state.